dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java

```diff
@@ -169,6 +180,39 @@ public String rssFeed() {
         }
     }
 
+    @GET
+    @Produces({ "text/plain", "application/json" })
+    @Path("/runner_dependencies")
+    @ApiOperation(value = "Returns the file containing runner dependencies", response = String.class)
+    public Response getRunnerDependencies(
+            @ApiParam(value = "The Dockstore client version") @QueryParam("client_version") String clientVersion,
+            @ApiParam(value = "Python version, only relevant for the cwltool runner") @DefaultValue("2") @QueryParam("python_version") String pythonVersion,
+            @ApiParam(value = "The tool runner", allowableValues = "cwltool") @DefaultValue("cwltool") @QueryParam("runner") String runner,
+            @ApiParam(value = "Response type", allowableValues = "json, text") @DefaultValue("text") @QueryParam("output") String output,
+            @Context ContainerRequestContext containerRequestContext) {
+        if (!("cwltool").equals(runner)) {
+            return Response.noContent().build();
+        }
+        boolean unwrap = !("json").equals(output);
+        URL url;
+        String content;
+        try {
+            String fileVersion = PipHelper.convertSemVerToAvailableVersion(clientVersion);
+            if (pythonVersion.startsWith("3")) {
+                url = this.getClass().getClassLoader().getResource("requirements/" + fileVersion + "/requirements3.txt");
```

Comment:
A few comments:
1) Charset.defaultCharset can change based on the underlying OS, just use UTF-8 since we provide this file and know that works with UTF-8. 
2) The last bit can just be something like `"/requirements" + (pythonVersion.startsWith("3") ? "3" :""` to avoid the copy-paste here
3) The InputStreamReaders are being opened, but not closed http://tutorials.jenkov.com/java-io/inputstreamreader.html#closing-a-inputstreamreader 
You could go with a "try, close with resources" but I suggest just avoiding the stream reader altogether with something like https://google.github.io/guava/releases/23.0/api/docs/com/google/common/io/Files.html#asCharSource-java.io.File-java.nio.charset.Charset-